examples/ec2-instance-feature-demo/eip-nic-blueprint.yaml

```diff
@@ -0,0 +1,61 @@
+tosca_definitions_version: cloudify_dsl_1_3
+imports:
+  - http://www.getcloudify.org/spec/cloudify/4.5/types.yaml
+  - plugin:cloudify-aws-plugin
+inputs:
+  aws_access_key_id:
+    type: string
+    description: YOUR AWS ACCESS KEY ID
+    default: { get_secret: aws_access_key_id }
+  aws_secret_access_key:
+    type: string
+    description: YOUR AWS SECRET ACCESS KEY
+    default: { get_secret: aws_secret_access_key }
+  region_name:
+    type: string
+    description: YOUR AWS Region
+    default: { get_secret: aws_region_name }
+  ec2_region_endpoint:
+    type: string
+    description: YOUR AWS Region endpoint
+    default: { get_secret: ec2_region_endpoint }
+  availability_zone:
+    type: string
+    description: Your AWS Region availability_zone.
+    default: { get_secret: aws_availability_zone }
+  use_external_ip:
+    type: boolean
+    default: False
+  existing_ip:
+    type: string
+    default: ''
+  subnet_id:
+    type: string
+dsl_definitions:
+  client_config: &client_config
+    aws_access_key_id: { get_input: aws_access_key_id }
+    aws_secret_access_key: { get_input: aws_secret_access_key }
+    region_name: { get_input: aws_region_name }
+node_templates:
+  nic:
+    type: cloudify.nodes.aws.ec2.Interface
+    properties:
+      client_config: *client_config
+      resource_config:
+        kwargs:
+          Description: Test EIP/NICs.
+          SubnetId: { get_input: subnet_id }
+  eip:
+    type: cloudify.nodes.aws.ec2.ElasticIP
+    properties:
+      client_config: *client_config
+      use_external_resource: { get_input: use_external_ip }
+      resource_id: { get_input: existing_ip }
+    relationships:
+      - type: cloudify.relationships.depends_on
+        target: nic
```

Comment:
@EarthmanT how external ip will be connected to the interface on this example ?

Reply:
`aws.cloudify_aws.ec2.resources.elasticip.attach` during elastic IP start operation.